### PR TITLE
remove dependency on train_mnist.py script

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
Remove dependency on train_mnist.py script that is removed as part of commit fb73de7

Also created an issue for replacing the script with a new one. #18549 

This fixes CD pipeline failures.

### CODE REVIEW

**Comment by mseth10** [comment]
@eric-haibin-lin @leezu can you please review. Thanks!

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1453,7 +1453,7 @@ nightly_test_imagenet_inference() {
 nightly_test_image_classification() {
     set -ex
     export DMLC_LOG_STACK_TRACE_DEPTH=10
-    ./tests/nightly/test_image_classification.sh
+    # TODO: the test script needs to be added (18549)
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh:1456]: Please do a proper cleanup instead of just removing references..

Comment (mseth10) [reply ci/docker/runtime_functions.sh]: Addressed.

**Comment by mseth10** [comment]
@mxnet-bot run ci [centos-gpu]

**Comment by mseth10** [comment]
@leezu  please help merge!

**Pull request merged**